### ISSUE TITLE
how about 5000 alphabet?

### ISSUE THREAD

**Issue by Aurora11111** [meijieru/crnn.pytorch#159]
hello,have you tried use more alphabet?
I have followed your ways to trianed rcnn model with just 10 numbers,10 chinese chars,10 mixed chinese chars and numbers and lower alpha,these model all perform well.but,when I try to trained a 5000 alphabet model ,the loss can't converge,it's about 75,do you know how to solve it?

**Comment by ZJU-PLP** [comment]
@Aurora11111 您好,请问一下你的10 numbers,10 chinese chars,10 mixed chinese chars and numbers and lower alpha是怎样训练的啊?制作数据集是用的哪个文件,我数据集做好了,但是感觉有问题,你能不能帮我看一下啊,谢谢啦?另外方便留一下您的邮箱或者其他联系方式吗,感激不尽!

**Comment by Aurora11111** [comment]
@ZJU-PLP 
refference:
https://github.com/Aurora11111/synthdata-zh
https://github.com/Aurora11111/TextRecognitionDataGenerator

**Comment by ZJU-PLP** [comment]
@Aurora11111 感谢回复,请问您最后数据集是做成ldmb格式吗?具体是路径--图片名--lable的格式吗?

**Comment by Aurora11111** [comment]
@ZJU-PLP yes. 
生成lmdb的参数需是:图片路径list,标签list
且一一对应

**Comment by ZJU-PLP** [comment]
@Aurora11111 我只做好数据集后出现了这个问题,请问你有遇到过嘛,在网上查不到这个问题
![selection_029](https://user-images.githubusercontent.com/32757213/45613752-d4ef1100-ba99-11e8-93b8-83325c1d20bb.png)

**Comment by ZJU-PLP** [comment]
@Aurora11111 我做好数据集后训练时出现了这个问题,请问你有遇到过嘛,在网上查不到这个问题
selection_029
![selection_029](https://user-images.githubusercontent.com/32757213/45613770-e506f080-ba99-11e8-887b-7908e32aad11.png)

**Comment by Aurora11111** [comment]
@ZJU-PLP 
链接https://github.com/Aurora11111/TextRecognitionDataGenerator
生成数据集的时候用python2.7,你的图片生成有问题

**Comment by ZJU-PLP** [comment]
@Aurora11111 这上面写着是python3.x啊
![selection_030](https://user-images.githubusercontent.com/32757213/45614601-a45ca680-ba9c-11e8-927c-5caae3cc3a45.png)

**Comment by Aurora11111** [comment]
那个是生成图片的时候,生成lmdb的时候用的python2.x
![img_20180917_171250 1](https://user-images.githubusercontent.com/39044868/45614742-0a492e00-ba9d-11e8-990a-cb33bd041d4a.jpg)

**Comment by ZJU-PLP** [comment]
@Aurora11111 谢谢提醒,按照你建议的方法,用Python2.x生成的数据集,但是训练的时候出现了这个问题,请问你训练的时候也是配置的python2.x的环境吗?
![image](https://user-images.githubusercontent.com/32757213/45659119-6d7ea300-bb25-11e8-9625-fd22f12c9dd1.png)

**Comment by Aurora11111** [comment]
只有生成lmdb用的python2.x
你的datasets.py和utils.py也需要参照上面链接里的改一下

**Comment by ZJU-PLP** [comment]
@Aurora11111 好的,非常感谢

**Comment by ZJU-PLP** [comment]
@Aurora11111 好的,非常感谢,目前已经跑通了,开始正常训练了,请问你0~9数字最后的准确率达到了多少呢?迭代了多少个epoch啊?

**Comment by Aurora11111** [comment]
我的也就一个迭代不到就到1了

**Comment by ZJU-PLP** [comment]
@Aurora11111 这么快嘛,我目前跑了14个epoch,arc为0.53,不过数据集做的2W,估计没有你的多

**Comment by Aurora11111** [comment]
@ZJU-PLP 恩,你的数据集应该弄多一些

**Comment by ZJU-PLP** [comment]
@Aurora11111 run demo.py出现这个问题.但是用之前下载的模型跑没问题,你跑的时候有遇到过吗?
![image](https://user-images.githubusercontent.com/32757213/45667151-8c436080-bb4a-11e8-94f7-2a5ded1d2e8d.png)

**Comment by Aurora11111** [comment]
@ZJU-PLP 需要修改demo.py中的参数与训练中的参数一致!

**Comment by ZJU-PLP** [comment]
@Aurora11111 训练时隐藏层数为128,demo.py中的参数也为128,其他参数都一样啊
![image](https://user-images.githubusercontent.com/32757213/45668418-40df8100-bb4f-11e8-9abf-495c4aa0a8f0.png)
![image](https://user-images.githubusercontent.com/32757213/45668501-8d2ac100-bb4f-11e8-8066-6c8e63418fff.png)

**Comment by Aurora11111** [comment]
@ZJU-PLP 这个问题issue里有:
![img_20180918_143653](https://user-images.githubusercontent.com/39044868/45668797-a1bb8900-bb50-11e8-8215-a7efd40c6cc6.jpg)

**Comment by ZJU-PLP** [comment]
@Aurora11111 你说的是这个吧,按这个方法来试过,并没有解决
![image](https://user-images.githubusercontent.com/32757213/45668927-fa8b2180-bb50-11e8-9e02-a197c5de9e72.png)

**Comment by ZJU-PLP** [comment]
@Aurora11111 ok,按你最新的代码可以跑通,谢谢!

**Comment by ZJU-PLP** [comment]
@Aurora11111 你最后有测过其他图片吗?我训练的时候模型显示准确率到了90%多,用测试集的图片去做测试,发现准确率也还OK,我数据集是train(20万)和val(5wan).但是用现实场景中的图片做测试,**测试了好几张,完全是错的**,请问你有遇到过这个问题嘛?感觉这两张图片也很相似啊,左边是数据集(都是电子字体)中的图片,右边是真实图片截取的一部分,完全检测不出来,放一大张图片就更检测不出来了
检测结果:
![image](https://user-images.githubusercontent.com/32757213/45751183-30f59900-bc44-11e8-81e2-23fe053a6cd9.png)
![image](https://user-images.githubusercontent.com/32757213/45751497-14a62c00-bc45-11e8-8bd7-51a8c3ce1467.png)

**Comment by liuyiyiyiyi** [comment]
请问所有的类别都要用alphabet列出来吗，比如说中文常见字有上千个，那是不是得在设置alphabet参数时将这些都列出来？数目这么多如果都要列出来的话需要用什么方法呢？

**Comment by Aurora11111** [comment]
@liuyiyiyiyi 是的，https://github.com/Aurora11111/TextRecognitionDataGenerator/blob/master/TextRecognitionDataGenerator/dicts/cn.txt
这里面修改

**Comment by liuyiyiyiyi** [comment]
使用你提供的样本自动生成代码，https://github.com/Aurora11111/TextRecognitionDataGenerator，是否可以将字体改成想要的形式，比如说：白色正楷粗体

**Comment by Aurora11111** [comment]
你要改里面的font吧

**Comment by helium390** [comment]
@Aurora11111 Thank you for [DataGenerator](https://github.com/Aurora11111/TextRecognitionDataGenerator). Is it possible to change color of fonts?

**Comment by liuyiyiyiyi** [comment]
如果我是要对自然场景下的中文字进行识别，那我生成的数据是不是要尽量与该自然场景下的中文字相似，比如说：背景和字体，if I want to recognize Chineses character in natural scene,should I keep the generated data,like background and fonts,more similar to the data in natural scene?

**Comment by Aurora11111** [comment]
是的

**Comment by liuyiyiyiyi** [comment]
你好，我用从自然场景下抠下来的文本进行乐预测，预测出来的标签序列有我的alphabet中不包含的字符，我感觉这个很奇怪，请问你有碰到过这个问题吗

**Comment by liuyiyiyiyi** [comment]
你好，我用从自然场景下抠下来的文本进行预测，预测出来的标签序列有我的alphabet中不包含的字符，我感觉这个很奇怪，请问你有碰到过这个问题吗

**Comment by Aurora11111** [comment]
@liuyiyiyiyi 那你要自己做一个dictionary了

**Comment by liuyiyiyiyi** [comment]
做一个dictionary什么意思呢，我是没有使用dictionary的

**Comment by HouBiaoLiu** [comment]
@Aurora11111 请教一下如果我想c++借口调用这个模型 应该怎么做

**Comment by Aurora11111** [comment]
@HouBiaoLiu 没试过,但是我觉得可以写个C++调用python的接口

**Comment by Giggledefy** [comment]
@Aurora11111 我遇到了之前 @ZJU-PLP 遇到的那几个问题，我直接用了你的crnn-train-pytorch,但是我对数据集的数据结构表示迷惑。我通过你的getLmdb.py写出的数据list是类似于：
`/home/users/Desktop/data/img/IMG_3222.JPG 图片3222文本内容`
是这样么？可以给我一个样例么？

**Comment by Aurora11111** [comment]
@Giggledefy 是的

**Comment by Pratyeka** [comment]
> @Aurora11111 您好,请教您最后有测过其他图片吗?我训练的时候模型显示准确率到了90%多,用测试集的图片去做测试,发现准确率也还OK,我数据集是train(20万)和val(5wan).但是用现实场景中的图片做测试,**测试了好几张,完全是错的**,请问你有遇到过这个问题嘛?感觉这两张图片也很相似啊,左边是数据集(都是电子字体)中的图片,右边是真实图片截取的一部分,完全检测不出来,放一大张图片就更检测不出来了
> 检测结果:
> ![image](https://user-images.githubusercontent.com/32757213/45751183-30f59900-bc44-11e8-81e2-23fe053a6cd9.png)
> ![image](https://user-images.githubusercontent.com/32757213/45751497-14a62c00-bc45-11e8-8bd7-51a8c3ce1467.png)

碰到了同样的问题，问题在什么地方？

**Comment by Pratyeka** [comment]
> @Aurora11111 您好,请教您最后有测过其他图片吗?我训练的时候模型显示准确率到了90%多,用测试集的图片去做测试,发现准确率也还OK,我数据集是train(20万)和val(5wan).但是用现实场景中的图片做测试,**测试了好几张,完全是错的**,请问你有遇到过这个问题嘛?感觉这两张图片也很相似啊,左边是数据集(都是电子字体)中的图片,右边是真实图片截取的一部分,完全检测不出来,放一大张图片就更检测不出来了
> 检测结果:
> ![image](https://user-images.githubusercontent.com/32757213/45751183-30f59900-bc44-11e8-81e2-23fe053a6cd9.png)
> ![image](https://user-images.githubusercontent.com/32757213/45751497-14a62c00-bc45-11e8-8bd7-51a8c3ce1467.png)

您好，我也碰到了这样的问题，请指点对应的解决办法

**Comment by Aurora11111** [comment]
@Pratyeka reference :https://github.com/Aurora11111/TextRecognitionDataGenerator
